### DISCUSSION TITLE
"Please try again in 24 hours." error

### DISCUSSION THREAD

**Discussion by quazardous** [thomluther/anker-solix-api#142] [Q&A] [answered]
my power agent does this every 5 minutes but I've got an error  after a while

```python 
       myapi = api.AnkerSolixApi(
            common.user(),
            common.password(),
            common.country(),
            websession,
            _LOGGER,
        )

        await myapi.update_sites()
        await myapi.update_site_details()
        await myapi.update_device_details()
        await myapi.update_device_energy()
        # Stuff

```

```json
{"code":100053,"msg":"Please try again in 24 hours.","trace_id":"3ce7fe874145e759411ed598aacc03b5"}
```

what can I do to reduce / cache things ?

I mainly need 2 values
```python
home_load_power = int(myapi.sites[BACKBATT_SITEID]["home_load_power"])
back_battery_level = float(int(myapi.devices[BACKBATT_SN]["battery_energy"]) / int(myapi.devices[BACKBATT_SN]["battery_capacity"]))
```

after that I do a `set_device_parm`

```python
myapi.set_device_parm(
  siteId=FRONTBATT_SITEID,
  paramType=SolixParmType.SOLARBANK_SCHEDULE.value,
  paramData=param_data
)
```

**Comment by thomluther** [comment, marked as answer]
You have to use separate accounts for the App and the Api.
Only the system owner can change things, so if you want to change anything via Api, you have to use a shared account for the App. which can only monitor
Every time when another client (App or Api) is connecting to the cloud, the existing token for the account on the cloud server is no longer valid and must be renewed. I think that can be done 10 times a day, then you get an IP ban and have to wait 24h or force your router to reconnect to the internet and hopefully get another IP.
You can use the Api client as often as you want, since it uses the cached token. But as I said, this remains only valid on the server until another Api client (the App) connects with their own cached token, which is different of course and throws away the Api client token to renew the token for the App client.

**Comment by quazardous** [comment]
yes, I've set up a secondary account but I had to change things a few times so I had to switch to the owner account...

10 times and ban seams to be it... thx

**Comment by quazardous** [comment]
Got it to work with tor socks proxy :smile: and the library `aiohttp_socks`

```python

import aiohttp
from aiohttp_socks import ProxyConnector

...

   # something like socks5://host.docker.internal:9050
   TOR_PROXY = os.getenv("TOR_PROXY", "")

   if TOR_PROXY != "":
        CONSOLE.info(f"Using TOR proxy {TOR_PROXY}")
        connector = ProxyConnector.from_url(TOR_PROXY)
        websession = aiohttp.ClientSession(connector=connector)
    else:
        websession = aiohttp.ClientSession()

...

```

**Comment by GehirnMaske** [comment]
I probably have a similar problem. 
Error message: 
"(26161) Anker Api Error: Failed to request.
Response Text: {"code":26161,"msg":"Failed to request.","trace_id":"4eaf14ae582328fc001ebebf5a3a1fe3"}
<class 'api.errors.RequestError'>: (26161) Anker Api Error: Failed to request."

I tried to query data cyclically: Node-Red with Injection Node, every 10 seconds. I used the account that is registered in the app for this. The Python script that was called connected to my account each time. After about a minute, I stopped receiving data. Since then, I have been unable to connect to the cloud with the API. Even accessing the app with my mobile phone fails when I am connected to my Raspberry's VPN. 
Is it possible that I have some kind of IP ban? I can attach my Python script if you like. However, it was tested on my Windows PC and the Raspberry Pi.

**Comment by thomluther** [comment]
I don't know. Typically you get a corresponding message if your IP is banned for 24h.
This could be any sort of problem.
Each endpoint also has throttling, so too many calls from same IP to that endpoint within a minute will be rejected. Typically also with a corresponsing message that your IP has been recorded.

**Comment by thomluther** [comment]
As of end of July 2025, multiple clients can now use the same account in parallel since active tokens on server are no longer deleted upon login requests.
So there should not longer be much risk for a 24h IP ban upon too many login attempts, since reconnections will work with the un-expired tokens of client and server.

**Comment by GehirnMaske** [comment]
Ok, thanks for the info. The error only occurred when I queried the data at 10 second intervals using Node Red. The account information is permanently entered in the Python script. The error message there was still: too many failed logins. Account is restricted for 10 minutes. 
That was strange, as the account information was permanently written in my script
I will continue testing this evening

**Reply by GehirnMaske** [reply]
Resetting my router solved the problem temporarily.
